string stack_name
uint32 stack_len
obstacle[] stack_obstacles
